tests/e2e/specs/backend/price-filter.test.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * External dependencies
+ */
+import {
+	openDocumentSettingsSidebar,
+	switchUserToAdmin,
+	getAllBlocks,
+	getEditedPostContent,
+} from '@wordpress/e2e-test-utils';
+import {
+	clearAndFillInput,
+	visitBlockPage,
+} from '@woocommerce/blocks-test-utils';
+import { queries, getDocument } from 'pptr-testing-library';
+
+/**
+ * Internal dependencies
+ */
+import { insertBlockDontWaitForInsertClose } from '../../utils.js';
+
+const block = {
+	name: 'Filter Products by Price',
+	slug: 'woocommerce/price-filter',
+	class: '.wp-block-woocommerce-price-filter',
+};
+
+describe( `${ block.name } Block`, () => {
+	describe( 'after compatibility notice is dismissed', () => {
+		beforeAll( async () => {
+			await switchUserToAdmin();
+			await visitBlockPage( `${ block.name } Block` );
+		} );
+
+		it( 'can only be inserted once', async () => {
+			await insertBlockDontWaitForInsertClose( block.name );
+			expect( await getAllBlocks() ).toHaveLength( 1 );
+		} );
+
+		it( 'renders without crashing', async () => {
+			await expect( page ).toRenderBlock( block );
+		} );
+
+		it( "allows changing the block's title", async () => {
+			const document = await getDocument( page );
+			// we focus on the block
+			await page.click( block.class );
+			await openDocumentSettingsSidebar();
+			await clearAndFillInput(
+				'.wp-block[data-type="woocommerce/price-filter"] textarea.wc-block-editor-components-title',
+				'New Title'
+			);
+			await page.click( block.class );
+
+			// Change title to h6.
+			const heading6Button = await queries.getByLabelText(
+				document,
+				/Heading 6/i
+			);
+			await heading6Button.click();
+			await expect(
+				page
+			).toMatchElement(
+				'.wp-block[data-type="woocommerce/price-filter"] h6 textarea',
+				{ text: 'New Title' }
+			);
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+			await clearAndFillInput(
+				'.wp-block[data-type="woocommerce/price-filter"] textarea.wc-block-editor-components-title',
+				'Filter by price'
+			);
+			// Change title to h3.
+			const heading3Button = await queries.getByLabelText(
+				document,
+				/Heading 3/i
+			);
+			await heading3Button.click();
+			await expect(
+				page
+			).not.toMatchElement(
+				'.wp-block[data-type="woocommerce/price-filter"] h6 textarea',
+				{ text: 'New Title' }
+			);
+		} );
```

Comment:
Resetting the block back to its initial state after will get painful quickly. Might be worth exploring this in a follow-up?
https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2783